### PR TITLE
Add missing `person` builder config to LSUN dataset.

### PR DESCRIPTION
**Pull request by AdamHillier**
The LSUN dataset on TDFS includes all [10 available scene categories](http://dl.yf.io/lsun/scenes/) and 19 out of the [20 available object categories](http://dl.yf.io/lsun/objects/).

The missing object category is `person` - likely omitted because it is by far the largest.

This PR adds the person category, which makes the LSUN dataset complete.

### CODE REVIEW

File: tensorflow_datasets/image/lsun.py
Comment (NikhilBartwal) [inline tensorflow_datasets/image/lsun.py:100]: Since, the dataset now has a completely new `BuilderConfig`, I think it would be better to version it `3.1.0`

File: tensorflow_datasets/url_checksums/lsun.txt
Comment (NikhilBartwal) [inline tensorflow_datasets/url_checksums/lsun.txt:35]: Have you generated the `checksums` locally on your end? If yes, could you please us tell a bit about the hardware and time utilized in the dataset generation. Since it is a >500GBs dataset, it might be a better option to use Apache Beam here.

**Review by NikhilBartwal — APPROVED** [review]
Thank you so much for the contribution @AdamHillier !

File: tensorflow_datasets/image/lsun.py
Comment (AdamHillier) [reply tensorflow_datasets/image/lsun.py]: Sure, done in 6686f37

File: tensorflow_datasets/url_checksums/lsun.txt
Comment (AdamHillier) [reply tensorflow_datasets/url_checksums/lsun.txt:35]: I did generate the checksums myself, yes.

For reference, the person category zip file is ~477 GB to download, and the generated TFRecords add up to 439 GB.

I actually used a GCP virtual machine to do it, one with: 32 vCPU cores, 32 GB RAM, 4 TB disk. However, I only did this because my local internet connection is slow. The overall process took around 15 hours, where the vast majority of time was taken by the initial download phase - only the last 90 mins was dataset generation.

I monitored resource usage it was always (during both the download phase and data generation phase) using < 20% CPU and < 10 GB RAM. Also, 4 TB disk wasn't required - at the end I had > 50% disk space remaining.

Comment (NikhilBartwal) [reply tensorflow_datasets/url_checksums/lsun.txt:35]: Thank you so much for the info @AdamHillier!
As per the given stats (evidently the low CPU usage), I think it would be better to use `Apache Beam` during dataset generation, instead of a casual loop as we might benefit from the underlying parallelization when generating the dataset in a distributed environment like GC Dataflow.
Please feel free to add the required changes for utilizing `Beam` during dataset generation as demonstrated [here](https://www.tensorflow.org/datasets/beam_datasets#instructions).

Also, it would be great if you could share the generated dataset's `dataset_info.json` file as it will help with updating the dataset documentation on the TFDS catalog page. Thanks!

File: tensorflow_datasets/image/lsun.py
Comment (NikhilBartwal) [inline tensorflow_datasets/image/lsun.py:162]: I'm not really sure about changing the structure of the dataset by adding in a new `id` feature as it might cause some issues for the previous users. 

Also, since the different configs consist of images belonging to the same category (of each config), I don't think the `id` feature would be required. Let me know your thoughts on this.

File: tensorflow_datasets/url_checksums/lsun.txt
Comment (AdamHillier) [reply tensorflow_datasets/url_checksums/lsun.txt:35]: Ah sorry for the delay - here is the `dataset_info.json`: https://gist.github.com/AdamHillier/5a5e9ccf9bc801a9de11ca5114efcb26.

Comment (AdamHillier) [reply tensorflow_datasets/url_checksums/lsun.txt:35]: For Apache Beam, would you like that to apply only to the `person` builder config or to all configs?

Comment (NikhilBartwal) [reply tensorflow_datasets/url_checksums/lsun.txt:35]: Thank you so much for the json file!
Since, majority of the configs are >100GB, I think it would be better to use `Beam` for all the datasets

File: tensorflow_datasets/image/lsun.py
Comment (AdamHillier) [reply tensorflow_datasets/image/lsun.py:162]: Why do you think adding a feature would break an existing use case? Wouldn't an existing user just ignore the presence of the additional feature?

For my own use case, I need to have the `id` feature, because I want to be able to perform certain per-image analysis tasks -- basically, I want to be able to associate arbitrary additional metadata with specific images, and having a static per-image identifier makes that much easier.

Comment (NikhilBartwal) [reply tensorflow_datasets/image/lsun.py:162]: Since, the dataset does not currently have supervised keys (which would have returned a tuple, instead of a dict ), I think you are right that the new feature would only add as an additional for the existing user base.

In that case, I think it would be good to add the supervised keys now (as `('id, 'image')`) as shown [here](https://www.tensorflow.org/datasets/add_dataset#_info_dataset_metadata).

I'll send a PR to update the required documentation once this PR gets approved from a maintainer or merged. Thanks!

**Review by Conchylicultor — APPROVED** [review]
Looks good. Thank you

**Pull request merged**
